### PR TITLE
ON-40638 # capture error where permissions are not correct

### PR DESCRIPTION
**Pull request by divporter**

### CODE REVIEW

File: src/form-elements/FormElementGoogleAddress.tsx [hunk 1]
```diff
@@ -62,27 +62,35 @@ function FormElementGoogleAddress({
...
-                reject('Google Places service not available')
-                return
+              switch (status) {
+                case google.maps.places.PlacesServiceStatus.ZERO_RESULTS: {
+                  resolve([])
+                  break
+                }
+                case google.maps.places.PlacesServiceStatus.REQUEST_DENIED: {
+                  reject(
+                    'Google Maps API key has not been configured correctly',
```
Comment (mymattcarroll) [inline src/form-elements/FormElementGoogleAddress.tsx:72]: bad practice to reject/throw anything other than an `Error`.

Comment (divporter) [reply src/form-elements/FormElementGoogleAddress.tsx:72]: yah fair enough, fixing

File: src/form-elements/FormElementGoogleAddress.tsx [hunk 2]
```diff
@@ -62,27 +62,39 @@ function FormElementGoogleAddress({
...
           )
         }).catch((e) => {
           if (!abortSignal.aborted) {
             Sentry.captureException(e)
           }
           throw new OneBlinkAppsError(
-            'An unknown error has occurred. Please contact support if the problem persists.',
-            { originalError: e },
+            e instanceof Error ? e.message : e.toString(),
           )
```
Comment (mymattcarroll) [inline src/form-elements/FormElementGoogleAddress.tsx:98]: Should just be 

```
`throw e`
```

File: src/form-elements/FormElementGoogleAddress.tsx [hunk 3]
```diff
@@ -162,7 +176,9 @@ function FormElementGoogleAddress({
         if (isMounted.current) {
           setError(
             new OneBlinkAppsError(
```
Comment (mymattcarroll) [inline src/form-elements/FormElementGoogleAddress.tsx:178]: same here,

**Pull request merged**